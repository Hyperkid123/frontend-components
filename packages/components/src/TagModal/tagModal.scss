.ins-c-tag-modal {
  height: calc(var(--pf-v5-global--spacer--4xl) + var(--pf-v5-global--breakpoint--md));
}
