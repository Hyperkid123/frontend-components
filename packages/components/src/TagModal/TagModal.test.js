import React from 'react';
import TagModal from './TagModal';
import { render } from '@testing-library/react';

describe('TagCount component', () => {
  it('Render the modal open with row of tags', () => {
    const { container } = render(
      <TagModal
        loaded
        isOpen={true}
        systemName={'paul.localhost.com'}
        rows={[
          ['key', 'value'],
          ['thing', 'otherthing'],
        ]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('Render the modal with a child component', () => {
    const { container } = render(
      <TagModal loaded isOpen={true} systemName={'paul.localhost.com'}>
        <h1>I am a child component</h1>
      </TagModal>
    );
    expect(container).toMatchSnapshot();
  });

  it('Should render modal with disabled button', () => {
    const { container } = render(
      <TagModal
        loaded
        isOpen={true}
        systemName={'paul.localhost.com'}
        rows={[
          ['key', 'value'],
          ['thing', 'otherthing'],
        ]}
        onApply={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render modal with enabled button', () => {
    const { container } = render(
      <TagModal
        loaded
        isOpen={true}
        systemName={'paul.localhost.com'}
        rows={[
          ['key', 'value'],
          ['thing', 'otherthing'],
        ]}
        selected={['thing']}
        onApply={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});

describe('Two tables', () => {
  it('should render two tables in loading state', () => {
    const { container } = render(
      <TagModal
        isOpen={true}
        systemName={'paul.localhost.com'}
        tabNames={['something', 'another']}
        rows={[[['something']], [['another']]]}
        columns={[[{ title: 'one' }], [{ title: 'two' }]]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render two tables', () => {
    const { container } = render(
      <TagModal
        isOpen={true}
        systemName={'paul.localhost.com'}
        tabNames={['something', 'another']}
        loaded={[true, true]}
        rows={[[['something']], [['another']]]}
        columns={[[{ title: 'one' }], [{ title: 'two' }]]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('Should render modal with disabled button', () => {
    const { container } = render(
      <TagModal
        isOpen={true}
        systemName={'paul.localhost.com'}
        tabNames={['something', 'another']}
        loaded={[true, true]}
        rows={[[['something']], [['another']]]}
        columns={[[{ title: 'one' }], [{ title: 'two' }]]}
        onApply={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render modal with enabled button', () => {
    const { container } = render(
      <TagModal
        isOpen={true}
        systemName={'paul.localhost.com'}
        tabNames={['something', 'another']}
        loaded={[true, true]}
        rows={[[['something']], [['another']]]}
        columns={[[{ title: 'one' }], [{ title: 'two' }]]}
        selected={[['thing']]}
        onApply={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
