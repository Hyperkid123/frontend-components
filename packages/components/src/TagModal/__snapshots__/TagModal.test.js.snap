// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCount component Render the modal open with row of tags 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`TagCount component Render the modal with a child component 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`TagCount component Should render modal with disabled button 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`TagCount component should render modal with enabled button 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Two tables Should render modal with disabled button 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Two tables should render modal with enabled button 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Two tables should render two tables 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Two tables should render two tables in loading state 1`] = `
<div
  aria-hidden="true"
/>
`;
